compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -3636,7 +3737,9 @@ private STNode parseQualifiedIdentifier(STNode identifier, boolean isInCondition
                 addInvalidTokenToNextToken(errorHandler.consumeInvalidToken());
                 return parseQualifiedIdentifier(identifier, isInConditionalExpr);
             default:
-                if (isInConditionalExpr) {
+                if (isInConditionalExpr && isPredeclaredPrefix) {
+                    return identifier;
```

Comment:
```suggestion
                    return createBuiltinSimpleNameReference(identifier);
```
Will this solve the problem we discussed offline?

Reply:
No, if `isPredeclaredPrefix` is true then identifier is already a BuiltinSimpleNameReference

Reply:
I mean we don't create the wrapper before calling the method. Instead we create the wrapper here?